streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -41,18 +50,25 @@
     private final Producer<byte[], byte[]> producer;
     private final Map<TopicPartition, Long> offsets;
     private final String logPrefix;
+    private final ProductionExceptionHandler productionExceptionHandler;
 
     private final static String LOG_MESSAGE = "Error sending record (key {} value {} timestamp {}) to topic {} due to {}; " +
         "No more records will be sent and no more offsets will be recorded for this task.";
     private final static String EXCEPTION_MESSAGE = "%sAbort sending since %s with a previous record (key %s value %s timestamp %d) to topic %s due to %s";
     private final static String PARAMETER_HINT = "\nYou can increase producer parameter `retries` and `retry.backoff.ms` to avoid this error.";
+    private final static String HANDLER_CONTINUED_MESSAGE = "Error sending records (key {} value {} timestamp {}) to topic {} due to {}; " +
+        "The exception handler chose to CONTINUE processing in spite of this error.";
```

Comment:
super nit: instead of `exception handler chose to CONTINUE` maybe `exception handler is configured to CONTINUE` , but this could just be a personal preference.

Reply:
I prefer the more active `chose` because the `ProductionExceptionHandler` will take in the exception and make a decision on what to do for that particular exception. My expectation is that folks will white-list exceptions to ignore as they make sense for their implementation case.